### PR TITLE
Add readonly and toolset request handlers

### PR DESCRIPTION
**Pull request by kerobbi**
## Summary

Adds support for `/readonly`, `/x/{toolset}` and `/x/{toolset}/readonly`.

## Why

- Part of https://github.com/github/github-mcp-server/pull/1849

## What changed

- Added context helpers for readonly mode, toolsets and tools
- Added `withReadonly` and `withToolset` route middleware 
- Registered `/readonly`, `/x/{toolset}` and `/x/{toolset}/readonly` routes
- Added `WithRequestConfig` middleware to extract MCP headers into context and refactored `InventoryFiltersForRequest` to primarily read from context instead of headers
- Added tests for inventory filtering and header parsing

## MCP impact

- [x] No tool or API changes
- [ ] Tool schema or behavior changed
- [ ] New tool added

## Prompts tested (tool changes only)

- 

## Security / limits

- [x] No security or limits impact
- [ ] Auth / permissions considered
- [ ] Data exposure, filtering, or token/size limits considered

## Tool renaming
- [ ] I am renaming tools as part of this PR (e.g. a part of a consolidation effort)
   - [ ] I have added the new tool aliases in `deprecated_tool_aliases.go` 
- [x] I am not renaming tools as part of this PR

Note: if you're renaming tools, you *must* add the tool aliases. For more information on how to do so, please refer to the [official docs](https://github.com/github/github-mcp-server/blob/main/docs/tool-renaming.md).

## Lint & tests

- [x] Linted locally with `./script/lint`
- [x] Tested locally with `./script/test`

## Docs

- [x] Not needed
- [ ] Updated (README / docs / examples)

### CODE REVIEW

File: pkg/context/request.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	return context.WithValue(ctx, toolsetCtxKey{}, toolset)
+}
+
+// GetToolset retrieves the active toolset from the context
+func GetToolset(ctx context.Context) string {
+	if toolset, ok := ctx.Value(toolsetCtxKey{}).(string); ok {
+		return toolset
+	}
+	return ""
+}
```
Comment (omgitsads) [inline pkg/context/request.go]: This should be `WithToolsets(ctx context.Context, toolsets []string) context.Context` and `GetToolsets(ctx context.Context) []string` since you can set multiple via the header.

You can then just set with `ghcontext.WithToolset(r.Context(), []string{toolset})` in the handler.

File: pkg/http/handler.go [hunk 2]
```diff
@@ -124,25 +148,38 @@ func DefaultInventoryFactory(cfg *HTTPServerConfig, t translations.TranslationHe
...
-func InventoryFiltersForRequestHeaders(r *http.Request, builder *inventory.Builder) *inventory.Builder {
-	if r.Header.Get(headers.MCPReadOnlyHeader) != "" {
+// InventoryFiltersForRequest applies inventory filters from request context and headers
+// Whitespace is trimmed from comma-separated values; empty values are ignored
+// Route configuration (context) takes precedence over headers for toolsets
+func InventoryFiltersForRequest(r *http.Request, builder *inventory.Builder) *inventory.Builder {
+	ctx := r.Context()
+
+	// Enable readonly mode if set in context or via header
+	if ghcontext.IsReadonly(ctx) || relaxedParseBool(r.Header.Get(headers.MCPReadOnlyHeader)) {
```
Comment (omgitsads) [inline pkg/http/handler.go]: I think we pull out the header parsing to a HTTP middleware that sets `WithReadonly`, and just have one method of assessing this. Same with Toolsets.

File: pkg/http/server.go [hunk 3]
```diff
@@ -98,6 +99,7 @@ func RunHTTPServer(cfg HTTPServerConfig) error {
 	r := chi.NewRouter()
 
 	handler := NewHTTPMcpHandler(&cfg, deps, t, logger)
+	r.Use(middleware.WithRequestConfig)
```
Comment (omgitsads) [inline pkg/http/server.go]: I think we need to move this into `RegisterRoutes` and potentially rename that or provide a `RegisterMiddleware` func to handle this.

File: pkg/http/middleware/request_config.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+// relaxedParseBool parses a string into a boolean value, treating various
+// common false values or empty strings as false, and everything else as true.
+// It is case-insensitive and trims whitespace.
+func relaxedParseBool(s string) bool {
+	s = strings.TrimSpace(strings.ToLower(s))
+	falseValues := []string{"", "false", "0", "no", "off", "n", "f"}
+	return !slices.Contains(falseValues, s)
+}
```
[the remark this answers is not in the document]
Comment (kerobbi) [reply pkg/http/middleware/request_config.go:40]: Valid but intentional, already documented.

File: pkg/context/request.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	return context.WithValue(ctx, toolsetsCtxKey{}, toolsets)
+}
+
+// GetToolsets retrieves the active toolsets from the context
+func GetToolsets(ctx context.Context) []string {
+	if toolsets, ok := ctx.Value(toolsetsCtxKey{}).([]string); ok {
+		return toolsets
+	}
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (kerobbi) [reply pkg/context/request.go:35]: Testing `InventoryFiltersForRequest` covers these already.

File: pkg/http/handler.go [hunk 6]
```diff
@@ -76,8 +76,32 @@ func NewHTTPMcpHandler(cfg *HTTPServerConfig,
...
+}
+
+// withToolset is middleware that extracts the toolset from the URL and sets it in the request context
+func withToolset(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		toolset := chi.URLParam(r, "toolset")
+		ctx := ghcontext.WithToolsets(r.Context(), []string{toolset})
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
 }
```
[the remark this answers is not in the document]
Comment (kerobbi) [reply pkg/http/handler.go:110]: Tested indirectly through `TestInventoryFiltersForRequest`.

File: pkg/http/handler_test.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+			available := inv.AvailableTools(context.Background())
+			toolNames := make([]string, len(available))
+			for i, tool := range available {
+				toolNames[i] = tool.Tool.Name
+			}
+
+			assert.ElementsMatch(t, tt.expectedTools, toolNames)
+		})
+	}
```
[the remark this answers is not in the document]
Comment (kerobbi) [reply pkg/http/handler_test.go:100]: Both header and route set readonly to true. There aren't really any edge cases to test.

File: pkg/http/middleware/request_config.go [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+			ctx = ghcontext.WithReadonly(ctx, true)
+		}
+
+		if toolsets := headers.ParseCommaSeparated(r.Header.Get(headers.MCPToolsetsHeader)); len(toolsets) > 0 {
+			ctx = ghcontext.WithToolsets(ctx, toolsets)
+		}
+
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
```
[the remark this answers is not in the document]
Comment (kerobbi) [reply pkg/http/middleware/request_config.go:31]: This is a relatively simple header -> context mapping, tested indirectly through `TestInventoryFiltersForRequest`.
